### PR TITLE
[MARVP-33] Implement app feedback form

### PR DESCRIPTION
**Pull request by vijaysamanthapudi**
Related: https://konysolutions.atlassian.net/browse/MARVP-33

### CODE REVIEW

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+define({
+
+    //Type your controller code here
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Remove the above line and spaces in the file

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    selectedImage: "starselected.png",
+    unselectedImage: "starunselected.png",
+    objectServiceL: "KonyAccelerateStorageService",
+    dataModelObject: "opinion_answers",
+    ratingQuestionId: "11",
+    feedbackQuestionId: "12",
+    opinion_id: 18,
+    /**
+     * @function onClickOfRatingPrompt
+     * @scope private
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: For the scope of the function, according to the standard, we follow there is no need for mentioning scope separately. Just add "@private" and move that to the last.

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        };
+        feedbacks.push(rating);
+        feedbacks.push(feedbackComment);
+        var batch = {
+            "records": feedbacks
+        };
+        this.createRecord(this.objectServiceL, this.dataModelObject, batch, this.successInStoreFeedback.bind(this), this.failInStoreFeedback.bind(this));
+
+    },
+    /**
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Please move the createRecord function to the "obejctServiceOperation.js" file which is a global module.

Comment (krishk95) [reply KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: For the scope of the function, according to the standard, we follow there is no need for mentioning scope separately. Just add "@Private" and move that to the last.

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+     * @function failInStoreFeedback
+     * @scope private
+     * @description this function is the failure callback for the createRecord function.
+     * @param error - contains the info of the error object.
+     */
+    failInStoreFeedback: function(error) {
+        alert("failure in storing the feedback");
+    }
+
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js:86]: Remove the spaces

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+});
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Add a single line space after the completion of the file

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    ratingQuestionId: "11",
+    feedbackQuestionId: "12",
+    opinion_id: 18,
+    /**
+     * @function onClickOfRatingPrompt
+     * @scope private
+     * @description this function is invoked to toggle the selected and unselected images.
+     * @param eventObject - contains the info of the widget which is clicked.
+     */
+    onClickOfRatingPrompt: function(eventObject) {
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js:14]: Please remove the focus skin for the button

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
+define({
+
+    //Type your controller code here 
+    selectedImage: "starselected.png",
+    unselectedImage: "starunselected.png",
+    objectServiceL: "KonyAccelerateStorageService",
```
Comment (vigneshwarankarunanithi) [inline KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Use Constants in global constants file for service name and object name (add if not present)

Comment (vijaysamanthapudi) [reply KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: ObjectService and data model object will be moved to constants. Other keys are specific to form.

Comment (krishk95) [reply KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: That will be useful if we want to use that in some other form

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js (hunk 1 above)
Comment (vijaysamanthapudi) [reply KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Removed the commented text

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js (hunk 3 above)
Comment (vijaysamanthapudi) [reply KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Moved to Global module

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js (hunk 5 above)
Comment (vijaysamanthapudi) [reply KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Added New line to controller.

File: KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js (hunk 7 above)
Comment (vijaysamanthapudi) [reply KonyAccelerate/controllers/mobile/frmFeedbackController/frmFeedbackController.js]: Moved to Constants file.

**Pull request merged**